### ISSUE TITLE
Scope Matching in Task Configuration Manager Is Brittle

### ISSUE THREAD

**Issue by colin-grant-work** [eclipse-theia/theia#8169]
### Bug Description:

To use methods like `.setTaskConfigurations` on the `TaskConfigurationManager`, you must supply a `TaskConfigurationScope` `string` or enum value. If you supply a `string` specifying a folder scope, it *must* be the stringified URI (including `file://` prefix) of the scope. That is a consequence of finding an editor model from the `.getModel` method:

https://github.com/eclipse-theia/theia/blob/a1ea2307d50565944b42f74ba3797698697ec897/packages/task/src/browser/task-configuration-manager.ts#L163-L167

The check `model.scope === scope` without manipulating the input ensures that consumers of that method must supply exactly the correct string. It would seem more robust to convert strings to `URI` and compare fields. Otherwise, it might be useful to document, at least in a Javadoc, the correct form of the scope string that qualifies as a `TaskConfigurationScope` and that is required by `.getModel` and `.setTaskConfiguration`

### Steps to Reproduce:

1. Call `setTaskConfiguration` using a path (no `file://` prefix).
2. Observe that it fails (returns `false`).

### Additional Information

- Operating System: RHEL 7
- Theia Version: `master`

**Comment by tsmaeder** [comment]
Usually you would not construct such an URI by hand, but rather get it from some API. @colin-grant-work could you elaborate on the use case?

**Comment by colin-grant-work** [comment]
You're quite right that isn't built by hand, so it isn't really a big deal practically, but I lost a few minutes in a case where I had the path form of the `URI` available already and tried to use that in a call to `setTaskConfiguration`, but saw it fail. Since the type of `TaskConfigurationScope` is `string | TaskScope.Workspace | TaskScope.Global`, I had assumed that any string equivalent of a `URI` would work, and had to dig into the code to find out that it _had_ to be the prefixed version. It was an easy fix, then, but it at least seems worth documenting to save others the same few minutes' digging 😄.
